scripts-dev/lint.sh

```diff
@@ -7,15 +7,88 @@
 
 set -e
 
-if [ $# -ge 1 ]
-then
-  files=$*
+usage() {
+  echo
+  echo "Usage: $0 [-h] [-d] [paths...]"
+  echo
+  echo "-d"
+  echo "  Lint files that have changed since the last git commit."
+  echo
+  echo "  If paths are provided and this option is set, both provided paths and those"
+  echo "  that have changed since the last commit will be linted."
+  echo
+  echo "  If no paths are provided and this option is not set, all files will be linted."
+  echo
+  echo "  Note that paths with a file extension that is not '.py' will be excluded."
+  echo "-h"
+  echo "  Display this help text."
+}
+
+USING_DIFF=0
+files=()
+
+while getopts ":dh" opt; do
+  case $opt in
+    d)
+      USING_DIFF=1
+
+      # Check both staged and non-staged changes
```

Comment:
generally this soft of stuff would belong *after* the `getopts` loop - in a separate `if` statement if necessary. It's mostly convention, but means that you can handle cases where behaviour depends on more than one option.

Reply:
Good point - have moved the logic outside of the block.